File: data/multilang_comparison/program_go/program_go/src/func13.go
package main
import "math/rand"
var _ = rand.Intn
func Func13(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array134 *Array
   if pCounter > 0 {
       array134 = vars.Data[pCounter-1]
       pCounter--
       array134.RefC++
   } else {
       array134 = &Array{
           Size: 996,
           RefC: 1,
           Id: 134,
           Data: make([]uint32, 996),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array134
   var array135 *Array = Func17(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   if (PATH0 & 1) != 0 {
      var array138 *Array
      if pCounter > 0 {
          array138 = vars.Data[pCounter-1]
          pCounter--
          array138.RefC++
      } else {
          array138 = &Array{
              Size: 27,
              RefC: 1,
              Id: 138,
              Data: make([]uint32, 27),
          }
      }
      var loop41 int = 0;
      var loopLimit41 int = (50)/3 + 1;
      for ; loop41 < loopLimit41; loop41++ {
         for i := 0; i < array138.Size; i++ {
             array138.Data[i]--
         }
         for i := 0; i < array138.Size; i++ {
             if array138.Data[i] == 56 {
                 return array138
             }
         }
      }
      var params1 ArrayParam
      params1.Size = 3
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array134
      params1.Data[1] = array135
      params1.Data[2] = array138
      var array139 *Array = Func28(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array139.RefC--
      if array139.RefC == 0 {
          array139.Data = nil
          array139 = nil
      }
      array138.RefC--
      if array138.RefC == 0 {
          array138.Data = nil
          array138 = nil
      }
   
   } else {
      var params1 ArrayParam
      params1.Size = 2
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array134
      params1.Data[1] = array135
      var array140 *Array = Func22(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array140.RefC--
      if array140.RefC == 0 {
          array140.Data = nil
          array140 = nil
      }
   }
   array134.RefC--
   if array134.RefC == 0 {
       array134.Data = nil
       array134 = nil
   }
   return array135;
}

File: data/multilang_comparison/program_julia/program_julia/src/func13.jl
include("program_julia_head.jl")
function func13(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array134 = nothing
   if pCounter > 1
       array134 = vars.data[pCounter]
       pCounter -= 1
       array134.refC += 1
   else
       array134 = JArray(zeros(Int, 996), 996, 1, 134)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array134)
   params0.size += 1
   array135 = func17(params0, loopsFactor);
   if (PATH0 & 1) != 0
      array138 = nothing
      if pCounter > 1
          array138 = vars.data[pCounter]
          pCounter -= 1
          array138.refC += 1
      else
          array138 = JArray(zeros(Int, 27), 27, 1, 138)
      end
      loopLimit41 = (50)/3 + 1
      for i in 1:loopLimit41 
         for i in 0:(array138.size-1) 
            array138.data[i+1] -= 1
         end
         for i in 1:array138.size
             if array138.data[i] == 56
                     return array138
            end
         end
      end
      params1 = JArray_param{Int}()
      params1.size = 3
      push!(params1.data, array134)
      params1.size += 1
      push!(params1.data, array135)
      params1.size += 1
      push!(params1.data, array138)
      params1.size += 1
      array139 = func28(params1, get_path(), loopsFactor);
      array139.refC -= 1
      if array139.refC == 0
          empty!(array139.data)
          array139 = nothing
      end
      array138.refC -= 1
      if array138.refC == 0
          empty!(array138.data)
          array138 = nothing
      end
   
   else 
      params1 = JArray_param{Int}()
      params1.size = 2
      push!(params1.data, array134)
      params1.size += 1
      push!(params1.data, array135)
      params1.size += 1
      array140 = func22(params1, get_path(), loopsFactor);
      array140.refC -= 1
      if array140.refC == 0
          empty!(array140.data)
          array140 = nothing
      end
   end
   array134.refC -= 1
   if array134.refC == 0
       empty!(array134.data)
       array134 = nothing
   end
   return array135;
end

